### PR TITLE
Feat/profile

### PR DESCRIPTION
**Pull request by carlenee**
Detailed description of changes
What's New
Profile front end is integrated with back end

What's Updated
N/A

New Package Added
N/A 

Recommendations on how to test the changes
N / A

Checklist for developer
- [ ] The code builds clean without any errors or warnings
- [ ]  The code does not contain commented out code
- [ ]  The code does not log anything to console
- [ ]  I have added unit test(s) to cover new code and successfully executed ran it
- [ ]  I have thoroughly tested the new code and any adjacent features it may affect

Issue(s)
Need to implement mutations & fix state management\

Screenshots or videos (before and after if appropriate)
Before
N / A

After
N / A

### CODE REVIEW

File: src/app/login/page.tsx [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+          sx={{ textDecoration: 'underline', cursor: 'pointer' }}
+          onClick={handleRecoverPasswordClick}
+          data-testid="recover-account-link"
+        >
+          Recover Account
+        </Typography>
+        <button type="submit" className="bg-teal-50">
+          Log In
+        </button>
+      </form>
```
Comment (nataniadeandra) [inline src/app/login/page.tsx:171]: I recommend refactoring your code by extracting the form into a separate component, for example, `LoginForm.tsx`. This approach aligns with the best practices of Next.js and enhances the modularity of your code.

By doing so, you adhere to the ‘Don’t Repeat Yourself’ (DRY) principle, which is a fundamental concept in software development aimed at reducing redundancy. This principle makes your code more maintainable and less prone to errors.

Even if this form component is used only once, separating it improves the readability of your code. It makes it easier for others to understand the structure and functionality of your application.

In summary, this change would make your code more organized, reusable, and easier to understand, which are all important qualities in effective software development.

File: __tests__/profile/change-profile.test.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    fireEvent.change(fileInput, { target: { files: [file] } });
+
+    await waitFor(() => {
+      // Check for UI updates or state changes related to the file input
+      expect(screen.getByAltText('')).toHaveAttribute('src');
+    });
+  });
+
+  // Add more tests as needed
```
Comment (emiriko) [inline __tests__/profile/change-profile.test.tsx:68]: You should add more negative case for your feature. Because editing profile is pretty crucial, and testing for these case would be really beneficial.

File: src/redux/features/profileSlice.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  });
+
+  if (!response.ok) {
+    throw new Error('Failed to fetch profile');
+  }
+  const data = await response.json();
+  console.log('API Response:', data);
+
+  return data;
+});
```
Comment (emiriko) [inline src/redux/features/profileSlice.ts:39]: As catherine mentioned, it would be better to use the mutation provided by react-redux toolkit.

**Review by emiriko — CHANGES_REQUESTED** [review]
Hi carlene, good job there. Don't forget to change your base branch (to staging instead of main) because we implemented git flow on our project.

File: src/app/login/page.tsx (hunk 1 above)
Comment (emiriko) [reply src/app/login/page.tsx:171]: That's a really good point, by refactoring it will make the code look better and more readable.

**Review by JohannesSetiawan — APPROVED** [review]
Are you sure you want to merge into main branch ????

File: src/app/login/page.tsx (hunk 1 above)
Comment (nataniadeandra) [reply src/app/login/page.tsx:171]: Besides the file mentioned earlier, I believe numerous other files could benefit from refactoring. Additionally, remember to set the target branch for merging to `staging`, not `main`! 😊
